File: e2e/cypress-javascript/fixtures/providers.js
const { StepProvider } = require('@/providers/StepProvider');
const { ValidationProvider } = require('@/providers/ValidationProvider');
const { SupportProvider } = require('@/providers/SupportProvider');

const step = new StepProvider();
const validate = new ValidationProvider();
const support = new SupportProvider();

module.exports = { step, validate, support };

File: e2e/capybara-ruby/fixtures/providers.rb
# frozen_string_literal: true

require_relative '../providers/step_provider'
require_relative '../providers/support_provider'
require_relative '../providers/validation_provider'

module Providers
  module_function

  def step
    @step ||= StepProvider.new
  end

  def validate
    @validate ||= ValidationProvider.new
  end

  def support
    @support ||= SupportProvider.new
  end

  def reset!
    @step = nil
    @validate = nil
    @support = nil
  end
end

step = Providers.method(:step)
validate = Providers.method(:validate)
support = Providers.method(:support)
